internal/olm/operator/registry/operator_installer_test.go

```diff
@@ -16,22 +16,233 @@ package registry
+	Describe("NewOperatorInstaller", func() {
+		It("should create an OperatorInstaller", func() {
+			kubeconfigPath := os.Getenv(k8sutil.KubeConfigEnvVar)
+			cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
+			// TODO(asmacdo) Is there anything I can test here that wouldnt be caught by the compiler?
+			// TODO(asmacdo)It doesnt even throw an err.
+			_ = NewOperatorInstaller(cfg)
+		})
+	})
+
+	Describe("createSubscription", func() {
+		var (
+			oi  *OperatorInstaller
+			sch *runtime.Scheme
+		)
+		BeforeEach(func() {
+			// Setup and fake client
+			kubeconfigPath := os.Getenv(k8sutil.KubeConfigEnvVar)
+			cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
+			sch = runtime.NewScheme()
+			Expect(v1.AddToScheme(sch)).To(Succeed())
+			Expect(v1alpha1.AddToScheme(sch)).To(Succeed())
+			cfg.Client = fake.NewFakeClientWithScheme(sch)
+
+			// Create OperatorInstaller struct
+			oi = NewOperatorInstaller(cfg)
+			oi.CatalogCreator = FakeCatalogCreator{
+				client: cfg.Client,
+			}
+			oi.StartingCSV = "fakeName"
+			oi.cfg.Namespace = "fakeNS"
+		})
+
+		It("should create the subscription with the fake client", func() {
+			sub, err := oi.createSubscription(context.TODO(), "huzzah")
+			Expect(err).ToNot(HaveOccurred())
+
+			retSub := &v1alpha1.Subscription{}
+			subKey := types.NamespacedName{
+				Namespace: sub.GetNamespace(),
+				Name:      sub.GetName(),
+			}
+			err = oi.cfg.Client.Get(context.TODO(), subKey, retSub)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(retSub.GetName()).To(Equal(sub.GetName()))
+			Expect(retSub.GetNamespace()).To(Equal(sub.GetNamespace()))
+		})
+
+		It("should pass through any client errors (duplicate)", func() {
+
+			sub := newSubscription(oi.StartingCSV, oi.cfg.Namespace,
+				withPackageChannel(oi.PackageName, oi.Channel, oi.StartingCSV),
+				withCatalogSource("duplicate", oi.cfg.Namespace),
+				withInstallPlanApproval(v1alpha1.ApprovalManual))
+			oi.cfg.Client = fake.NewFakeClientWithScheme(sch, sub)
+
+			_, err := oi.createSubscription(context.TODO(), "duplicate")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("error creating subscription"))
+		})
+	})
+
+	// TODO(asmacdo) Since this uses the olm client can/should this be unit tested?
+	// Describe("getInstalledCSV", func() {
+	// 	})
+	// })
+	Describe("approveInstallPlan", func() {
+		var (
+			oi  *OperatorInstaller
+			sch *runtime.Scheme
+		)
+		BeforeEach(func() {
+			kubeconfigPath := os.Getenv(k8sutil.KubeConfigEnvVar)
+			cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
+			sch = runtime.NewScheme()
+			Expect(v1alpha1.AddToScheme(sch)).To(Succeed())
+			oi = NewOperatorInstaller(cfg)
+		})
+
+		It("should update the install plan", func() {
+			existingIp := &v1alpha1.InstallPlan{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fakeName",
+					Namespace: "fakeNS",
+				},
+			}
+			oi.cfg.Client = fake.NewFakeClientWithScheme(sch, existingIp)
```

Comment:
since you are not using existingIp anywhere else you can initialize the fake client inline:
```golang

		oi.cfg.Client = fake.NewFakeClientWithScheme(sch, 
			&v1alpha1.InstallPlan{
				ObjectMeta: metav1.ObjectMeta{
					Name:      "fakeName",
					Namespace: "fakeNS",
				},
			},
		)
```